reorg Stablecoin UC (#2132)
XRPLF/xrpl-dev-portal: content/use-cases/tokenization/stablecoin-issuer.md

3. Send a payment of stablecoins to the consumer up to the maximum amount in the trust line.

While anyone can issue a token with any currency code in the XRP Ledger, stablecoin value comes from the promise that it can be redeemed for a corresponding asset. Issuing a stablecoin might involve regulatory obligations, which vary by jurisdiction. For these reasons, an established, reputable business is most likely to succeed in issuing a stablecoin.

[![Stablecoin Workflow](img/uc-stablecoin-flow.png)](img/uc-stablecoin-flow.png)

There are many decisions to make and artifacts to generate as you prepare to release a new stablecoin. You can use the XRPL Foundation's [Self-assessment Questionnaire](https://foundation.xrpl.org/wp-content/uploads/2022/03/self_assessment_questionnaire_140322.pdf) as a starting point to gather and produce the necessary information for a successful launch.

## Choose the Type of Stablecoin

The first step is to decide the type of stablecoin you want to create. Your choice of stablecoin type might require additional steps, such as signing on financial or audit partners.

![Stablecoin](img/uc-stablecoin-stable-coin.png)

> There are five common types of stablecoin you can create on the XRPL: fiat-backed, crypto-backed, commodity-backed, financial instrument-backed, and non-collateralized. See [Stablecoin Types](stablecoin-types.html).

Reviewer: `stablecoin-types.html` appears to be a broken link. Is there a page missing?

Also, this is stretching the definition of a "stablecoin"; if it's non-collateralized, why/how should its value be stable? By definition, a stablecoin is supposed to be backed 1:1 with outside funds of equal value. Otherwise, it can be a token but not a stablecoin, IMO.

Author: Changed to _currency token_.

Alternatively, you can use an external node service provider like OpenNode. See [OpenNode](https://www.opennodecloud.com/).

## Sandbox Access

![Sandbox](img/uc-stablecoin-sandbox.png)

For testing purposes, you can implement, deploy, and trade your stablecoin on the XRPL Testnet or Devnet servers. Visit the XRP Faucets page to generate your test network credentials. Use the listed server URIs on that page to connect to and interact with your chosen test network. See [XRP Faucets](xrp-testnet-faucet.html).

## Stablecoin Settings

Before you mint your new stablecoin, you need to configure settings, some of which are immutable once you issue the first coin.

See [Stablecoin Settings](stablecoin-settings.html).

For more detail on configuration capbilities, see [Stablecoin Issuer Configuration](stablecoin-configuration.html).

## Asset Information

Publish standard information about your stablecoin to enable potential traders to verify the coin's stability.  

### Asset Nomenclature

> Choose a 3-character string for your currency code. Per ISO 4217, supranational currency codes begin with the letter _X_ for a currency that is not associated with a country. See [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217#X_currencies_(funds,_precious_metals,_supranationals,_other)).

Reviewer: The word "supranational" is obscure, and this sentence can be equally clear without it.

```suggestion
Choose a 3-character string for your currency code. Per ISO 4217, currencies that are not associated with a country should use currency codes that begin with the letter _X_. See [ISO 4217](https://en.wikipedia.org/wiki/ISO_4217#X_currencies_(funds,_precious_metals,_supranationals,_other)).
```

One misconception I've heard that might also be good to clear up here is that multiple issuers can choose the same currency code. (Some people assume that every issuer has to have their own unique currency code.) If you're issuing a stablecoin backed by a national fiat currency, it's actually preferable to use the official code for that currency, such as EUR.

Author: The term supranational is clear in context, if unfamiliar, and used in the linked material. Added info that currency codes don't have to be unique.

### xrp-ledger.toml 

> You can use the _Currencies_ table in an XRPL TOML file to provide additional information about your stablecoin. This makes the information about your cryptocurrency accessible in an expected place and format, and enhances transparency. See [xrp-ledger.toml File](xrp-ledger-toml.html#currencies).

Reviewer: It might be good to add a little more context about where this file comes from/goes. For example:

```suggestion
You can use the _Currencies_ table in an XRPL TOML file on your website to provide additional information about your stablecoin. This makes the information about your cryptocurrency accessible in an expected place and format, and enhances transparency. See [xrp-ledger.toml File](xrp-ledger-toml.html#currencies).
```

Author: Added the change.

- Various settings and metadata. Each of the two accounts can control its own settings on the trust line. Each side sets a limit on the trust line.

Each trust line is specific to a given currency code. Two accounts can have any number of trust lines between them for different currency codes, but only one shared trust line for any particular currency code.

See [Trust Lines](ft-index.html#trust-lines).

### Authorized Trust Lines

The Authorized Trust Lines feature enables issuers to create tokens that can only be held by accounts that the issuer specifically authorizes. This feature only applies to tokens, not XRP.

To use the Authorized Trust Lines feature, enable the `Require Auth` flag on your issuing account. While the setting is enabled, other accounts can only hold tokens you issue if you have authorized those accounts' trust lines to your issuing account.

See [Authorized Trust Lines](authorized-trust-lines.html).

### Freeze a Trust Line

If you issue tokens in the XRP Ledger, you can enable the _No Freeze_ setting to permanently limit your own ability to use the token freezing features of the XRP Ledger. (As a reminder, this only applies to issued tokens, not XRP.) 

> If you do not enable the _No Freeze_ setting, when an account shows suspicious activity or violates your institution's terms of use, you have the option of freezing the trust line while you resolve the issue.

Reviewer: I think it would make more sense to reverse the order of these paragraphs (and rephrase accordingly). For example:

> As a token issuer, when an account shows suspicious activity or violates your terms of use, you have the option of freezing your trust line to that account while you resolve the issue. (As a reminder, this only applies to issued tokens, not XRP.)
>
> To make your tokens more "cash-like" and provide assurances to users that you won't misuse these features, you can enable the _No Freeze_ setting to permanently limit your own ability to use the token freezing features of the XRP Ledger.
```

See [Freezing Tokens](freezes.html).

### Global Freeze

> If you see signs of suspicious activity, you can enact a global freeze on your account to prevent users from sending your tokens to each other and trading your token in the decentralized exchange. 

Reviewer: "suspicious activity" is vague here; there's all kinds of suspicious activity that doesn't justify a global freeze. Maybe it would be better as, "If you suspect that your own account or systems have been compromised," instead?

![Global Freeze](img/uc-stablecoin-global-freeze.png)

See [Enact Global Freeze](enact-global-freeze.html)

### Clawback

Clawback is an optional setting that you can choose before you begin to distribute your stablecoin. For regulatory purposes, some issuers _must_ have the ability to recover issued tokens after they are distributed to accounts. For example, if an issuer were to discover that tokens were sent to an account sanctioned for illegal activity, the issuer could recover, or _claw back_, the funds.

See [Clawback](clawback.html).

![Clawback](img/uc-stablecoin-clawback.png)

### Burn

> One way to manage the value of a stablecoin is to destroy, or _burn_ stablecoins. When you burn stablecoins, you reduce the number of tokens in circulation. That will typically increase the value of the remaining tokens as supply falls.

Reviewer: This goes back to my comment above about stretching the definition of a stablecoin. The value per unit of a stablecoin is supposed to be fixed with respect to its collateral, so burning stablecoins _must_ mean that there is less collateral, unless the coin has become un-pegged and is no longer a stablecoin.

> To destroy stablecoins, you can transfer them to a "black hole" account (an account address where no one knows the secret key).

Reviewer: This is misleading/incorrect. You cannot always transfer tokens to a black hole account (it would have to have created a trust line to your issuer before becoming a black hole). However, you _can_ always send tokens back to the issuing account, which directly reduces the number of tokens outstanding. That's burning the tokens in a more direct and literal sense.

However, there's something off about the bigger picture here. For a non-stablecoin token with a fixed total supply  and a blackholed issuer, it might make sense to talk about "burning" tokens. But for a stablecoin, it's weird; typically, the same process is called "redeeming" the tokens (for their off-ledger collateral value). The way a stablecoin (specifically) is supposed to work, the on-ledger supply fluctuates in response to increases or decreases in collateral (that is, deposits and withdrawals) and tokens are "burned" or "issued" regularly.

- Protect yourself against reversible deposits. XRP Ledger payments are irreversible, but many digital payments are not. Scammers can abuse this to take their fiat money back by canceling a deposit after receiving tokens in the XRP Ledger.
- When sending into the XRP Ledger, always specify your issuing address as the issuer of the token. Otherwise, you might accidentally use paths that deliver the same currency issued by other addresses.
- Before sending a payment into the XRP Ledger, double check the cost of the payment. A payment from your operational address to a customer should not cost more than the destination amount plus any transfer fee you have set.
- Before processing a payment out of the XRP Ledger, make sure you know the customer's identity. This makes it harder for anonymous attackers to scam you. Most anti-money-laundering regulations require this anyway. This is especially important because the users sending money from the XRP Ledger could be different than the ones that initially received the money in the XRP Ledger.
- Follow the guidelines for [reliable transaction submission](#reliable-transaction-submission) when sending XRP Ledger transactions.

> - [Robustly monitor for incoming payments](#robustly-monitoring-for-payments), and read the correct amount. Don't mistakenly credit someone the full amount if they only sent a [partial payment](partial-payments.html).

Reviewer: It's imperative that the stablecoin issuer use case warn people about partial payments. There have been multiple instances of issuers losing thousands to millions of dollars because of an exploit of their systems caused by misunderstanding partial payments.

I would say it's even appropriate to add an entire section to the doc alerting issuers to the possibility of partial payments and linking to the doc.

Author: Added a partial payments section with link.

> The goal of reliably submitting transactions is to achieve the following two properties in a finite amount of time:

Reviewer: Seeing this in the new context, it's more obvious to me that this summary buries the lede on why you should care about reliable transaction submission. What it probably should say here is something like:

> Best practices for reliable transaction submission ensure that your transactions only execute once, within a finite and predictable amount of time, even if your systems suffer problems such as a power failure or network outage. Without taking proper care, it is possible that transactions could be delayed an indefinite amount of time, you might not realize that a payment to a customer has failed, or you might mistakenly send money again for a transaction that has already been settled.

For more information, see [Reliable Transaction Submission](reliable-transaction-submission.html).

### List on an XRPL Native DEX

> Decentralized exchanges are integral to the decentralized finance ecosystem. Listing your token on well known DEXs operating on XRPL will enhance its visibility and accessibility, thereby attracting more liquidity. Begin by placing sell offers on a suitable DEX, such as [Sologenic](https://sologenic.org/trade).

Reviewer: The wording here is a bit misleading: there is _one, global DEX_ built into the XRP Ledger. Sologenic and others are just _interfaces_ into the same shared system. (Offers placed on GateHub can be seen and matched on Sologenic, and vice-versa.)

Also, token issuers don't need to do anything to list on the DEX. They can, if they want, set up a trading account to place offers for their token. This guide should probably provide the advice that they use a separate account for trading than they do for issuing.
